### PR TITLE
feat: Receive notification on connection request

### PR DESCRIPTION
**Pull request by borichellow**
Needed for [task-1826](https://wearezeta.atlassian.net/browse/AR-1826)
----

# What's new in this PR?
Updated `GetNotificationsUseCase` to return notification about new Connection requests too

### CODE REVIEW

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/message/GetNotificationsUseCase.kt [hunk 1]
```diff
@@ -32,17 +34,20 @@ interface GetNotificationsUseCase {
...
  * @param messageMapper MessageMapper for mapping Message object into LocalNotificationMessage
- * @param publicUserMapper PublicUserMapper for mapping PublicUser object into LocalNotificationMessageAuthor
+ * @param localNotificationMessageMapper LocalNotificationMessageMapper for mapping PublicUser object into LocalNotificationMessageAuthor
  *
  * @return Flow<List<LocalNotificationConversation>> - Flow of Notification List that should be shown to the user.
  * That Flow emits everytime when the list is changed
  */
+@Suppress("LongParameterList")
 class GetNotificationsUseCaseImpl(
+    private val observeConnectionList: ObserveConnectionListUseCase,
```
Comment (Garzas) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/message/GetNotificationsUseCase.kt:50]: I don't know if there is some agreement here, but should we use use cases in other use cases? 😄

File: logic/src/commonTest/kotlin/com/wire/kalium/logic/feature/message/GetNotificationsUseCaseTest.kt [hunk 2]
```diff
@@ -341,7 +356,42 @@ class GetNotificationsUseCaseTest {
         }
     }
 
+    @Test
+    fun givenConnectionRequests_thenNotificationListWithConnectionRequestMessage() = runTest {
```
Comment (Garzas) [inline logic/src/commonTest/kotlin/com/wire/kalium/logic/feature/message/GetNotificationsUseCaseTest.kt:360]: Maybe write another test to hide connection request from notifications when some one cancel it?

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/message/GetNotificationsUseCase.kt (hunk 1 above)
Comment (borichellow) [reply logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/message/GetNotificationsUseCase.kt:50]: we already have such cases, like in `ObserveConversationsAndConnectionsUseCaseImpl` :)

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/message/GetNotificationsUseCase.kt [hunk 3]
```diff
@@ -210,24 +227,32 @@ class GetNotificationsUseCaseImpl(
...
+        MessageContent.MissedCall -> true
+        is MessageContent.Text -> true
+        is MessageContent.Calling -> false
+        is MessageContent.Asset -> true
+        is MessageContent.DeleteMessage -> false
+        is MessageContent.TextEdited -> false
+        is MessageContent.RestrictedAsset -> true
+        is MessageContent.DeleteForMe -> false
+        MessageContent.Empty -> false
+        MessageContent.Ignored -> false
```
[the remark this answers is not in the document]
Comment (borichellow) [reply logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/message/GetNotificationsUseCase.kt:254]: hmm, IMO 
```
    private fun isMessageContentSupportedInNotifications(message: Message): Boolean = when (message.content) {
        MessageContent.Empty, MessageContent.Ignored, is MessageContent.DeleteForMe, is MessageContent.Unknown,
        is MessageContent.TextEdited, is MessageContent.MemberChange, is MessageContent.Calling, is MessageContent.DeleteMessage -> false
        MessageContent.MissedCall, is MessageContent.Text, is MessageContent.Asset, is MessageContent.RestrictedAsset -> true
    }
```
looks not really cool :)

File: logic/src/commonTest/kotlin/com/wire/kalium/logic/feature/message/GetNotificationsUseCaseTest.kt (hunk 2 above)
Comment (borichellow) [reply logic/src/commonTest/kotlin/com/wire/kalium/logic/feature/message/GetNotificationsUseCaseTest.kt:360]: there is no need for such test, as for now all the ConnectionRequests that user should be notified about are got by DB-query. 
So there is nothing to test in that UseCase in fact

**Comment by Garzas** [comment]
LGTM 💪

**Pull request merged**